examples/platform/silabs/SiWx917/SiWxWifiInterface.cpp

```diff
@@ -328,9 +328,8 @@ sl_status_t InitiateScan()
     sl_wifi_scan_configuration_t wifi_scan_configuration = default_wifi_scan_configuration;
 
     ssid.length = wfx_rsi.sec.ssid_length;
-
-    // TODO: workaround because the string management with the null termination is flawed
-    chip::Platform::CopyString((char *) &ssid.value[0], ssid.length + 1, wfx_rsi.sec.ssid);
+    ssid.length = std::min<size_t>(wfx_rsi.sec.ssid_length, sizeof(ssid.value) - 1);
```

Comment:
Why are we setting `ssid.length` two lines in a row?  If the second line had used the value from the first, that would make sense, but it does not.

Reply:
Ohh didn't noticed that the assign was added not modified. Removed the first assignment.